python/ray/actor.py

```diff
@@ -129,34 +127,112 @@ def annotate_method(method):
+    ):
+        """Initialize an ActorMethodShell.
+
+        Args:
+            method_name: The name of the actor method.
+            num_returns: The default number of return values that the method
+                invocation should return. If None is given, it uses
+                DEFAULT_ACTOR_METHOD_NUM_RETURN_VALS for a normal actor task
+                and "streaming" for a generator task (when `is_generator` is True).
+            max_task_retries: Number of retries on method failure.
+            retry_exceptions: Boolean or list/tuple of exceptions to retry.
+            is_generator: True if the method is a generator.
+            generator_backpressure_num_objects: Generator-only config for backpressure.
+            enable_task_events: True if task events are enabled for this method.
+            decorator: Optional decorator for the method invocation.
+            signature: The signature of the actor method.
+            tensor_transport: The tensor transport protocol to use for the actor method.
+        """
+        self.method_name = method_name
+
+        # Default case.
+        if num_returns is None:
+            if is_generator:
+                num_returns = "streaming"
+            else:
+                num_returns = ray_constants.DEFAULT_ACTOR_METHOD_NUM_RETURN_VALS
+        self.num_returns = num_returns
+        self.max_task_retries = max_task_retries
+        self.retry_exceptions = retry_exceptions
+        self.is_generator = is_generator
+        self.generator_backpressure_num_objects = generator_backpressure_num_objects
+        self.enable_task_events = enable_task_events
+        self.decorator = decorator
+        self.signature = signature
+        self.tensor_transport = tensor_transport
+
+    def bind(self, actor_handle):
+        """
+        Produce a bound ActorMethod that holds a strong reference to actor_handle.
+        """
+        return ActorMethod(
+            actor_handle,
+            self.method_name,
+            self.num_returns,
+            self.max_task_retries,
+            self.retry_exceptions,
+            self.is_generator,
+            self.generator_backpressure_num_objects,
+            self.enable_task_events,
+            decorator=self.decorator,
+            signature=self.signature,
+            tensor_transport=self.tensor_transport,
+        )
+
+
 # Create objects to wrap method invocations. This is done so that we can
 # invoke methods with actor.method.remote() instead of actor.method().
 @PublicAPI
 class ActorMethod:
     """A class used to invoke an actor method.
 
-    Note: This class only keeps a weak ref to the actor, unless it has been
-    passed to a remote function. This avoids delays in GC of the actor.
+    Note: When bound (via ActorMethodShell.bind), each instance holds a
+    strong reference to its ActorHandle for the duration of the invocation,
+    preventing premature GC and simplifying lifecycle semantics.
```

Comment:
this docstring is public, and users should not know about the `ActorMethodShell` (it's purely an implementation detail)

either:

1. remove this sentence entirely, since we removed the unexpected behavior (preferred)
2. if you think it's necessary, document the user-facing behavior, which is that holding a reference to the actor method will keep the actor in scope